File: server/bench/fixtures/sample.cpp
#include <iostream>
#include <string>
#include <vector>
#include <optional>

namespace sample {

struct User {
    std::string name;
    int age;
};

class Repository {
public:
    explicit Repository(std::vector<User> users) : users_(std::move(users)) {}

    std::optional<User> find(const std::string &name) const {
        for (const auto &u : users_) {
            if (u.name == name) return u;
        }
        return std::nullopt;
    }

    std::size_t count() const { return users_.size(); }

private:
    std::vector<User> users_;
};

std::string greet(const User &u) {
    return "Hello, " + u.name + "!";
}

} // namespace sample

int main() {
    using namespace sample;
    Repository repo({{"alice", 30}, {"bob", 25}});
    if (auto u = repo.find("alice")) std::cout << greet(*u) << "\n";
    std::cout << "total: " << repo.count() << "\n";
    return 0;
}

File: server/bench/fixtures/sample.rs
use std::collections::HashMap;

trait Greeter {
    fn greet(&self, name: &str) -> String;
}

#[derive(Debug, Clone)]
pub struct User {
    pub name: String,
    pub age: u32,
}

pub enum Role {
    Admin,
    User,
    Guest,
}

pub struct Repository {
    users: HashMap<String, User>,
}

impl Repository {
    pub fn new() -> Self {
        Self { users: HashMap::new() }
    }

    pub fn add(&mut self, u: User) {
        self.users.insert(u.name.clone(), u);
    }

    pub fn find(&self, name: &str) -> Option<&User> {
        self.users.get(name)
    }

    pub fn count(&self) -> usize {
        self.users.len()
    }
}

impl Greeter for Repository {
    fn greet(&self, name: &str) -> String {
        format!("Hello, {}!", name)
    }
}

fn main() {
    let mut repo = Repository::new();
    repo.add(User { name: "alice".to_string(), age: 30 });
    repo.add(User { name: "bob".to_string(), age: 25 });
    if let Some(u) = repo.find("alice") {
        println!("{}", repo.greet(&u.name));
    }
    println!("total: {}", repo.count());
}
